api/src/infrastructure/kubernetes/infrastructure.rs

```diff
@@ -500,20 +494,26 @@ impl KubernetesInfrastructure {
         &self,
         app_name: &str,
         service: &'a DeployableService,
-    ) -> Result<HashMap<&'a String, PersistentVolumeClaim>, KubernetesInfrastructureError> {
-        let client = self.client().await?;
+    ) -> Option<HashMap<&'a String, PersistentVolumeClaim>> {
+        let client = match self.client().await.ok() {
```

Comment:
Why not using the `?` operator here. Or [`let … else { return … }?`](https://doc.rust-lang.org/beta/rust-by-example/flow_control/let_else.html).